include/swift/Serialization/SerializedModuleLoader.h

```diff
@@ -70,20 +71,30 @@ class SerializedModuleLoaderBase : public ModuleLoader {
   virtual std::error_code findModuleFilesInDirectory(
       AccessPathElem ModuleID, StringRef DirPath, StringRef ModuleFilename,
       StringRef ModuleDocFilename,
+      StringRef ModuleSourceInfoFilename,
       std::unique_ptr<llvm::MemoryBuffer> *ModuleBuffer,
-      std::unique_ptr<llvm::MemoryBuffer> *ModuleDocBuffer) = 0;
+      std::unique_ptr<llvm::MemoryBuffer> *ModuleDocBuffer,
+      std::unique_ptr<llvm::MemoryBuffer> *ModuleSourceInfoBuffer) = 0;
```

Comment:
These now-three files are passed together so often that I'm beginning to think we should have an abstraction to represent this grouping. `ModuleFiles<StringRef> ModuleFilenames`, `ModuleFiles<std::unique_ptr<llvm::MemoryBuffer> *> ModuleBuffers`.